MarketOrganizer.Api/Services/AuthService.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+using MarketOrganizer.Api.Interfaces;
+using MarketOrganizer.Data.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+using MarketOrganizer.Api.Helpers;
+using System.IdentityModel.Tokens.Jwt;
+using System.Text;
+using Microsoft.IdentityModel.Tokens;
+using System.Security.Claims;
+
+namespace MarketOrganizer.Api.Services
+{
+  public class AuthService : IAuthService<User>
+  {
+    private ItemsContext _context;
+    private readonly AppSettings _appSettings;
+
+    public AuthService(ItemsContext context, IOptions<AppSettings> appSettings)
+    {
+      _appSettings = appSettings.Value;
+      _context = context;
+    }
+
+    public User Login(string username, string password)
+    {
+      var user = _context.Users.SingleOrDefault(x => x.Username == username && x.Password == password);
+      if (user == null)
+      {
+        return null;
+      }
+      var tokenHandler = new JwtSecurityTokenHandler();
+      var key = Encoding.ASCII.GetBytes(_appSettings.Secret);
+      var tokenDescriptor = new SecurityTokenDescriptor
+      {
+        Subject = new ClaimsIdentity(new Claim[]
+          {
+                    new Claim(ClaimTypes.Name, user.Id.ToString())
+          }),
+        Expires = DateTime.UtcNow.AddMinutes(5),
+        SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+      };
+      var token = tokenHandler.CreateToken(tokenDescriptor);
+      user.Token = tokenHandler.WriteToken(token);
```

Comment:
yo te aconsejaria que todo lo que tiene que ver con lo del token lo pasaras a un metodo aparte
preguntate a ti mismo:
>Cuando veo esto a la primera pienso en un Login?

al ver todo lo del token como que se pierde un poco la intencion del login
```suggestion
    var token = GenerarToken();
    //... blababla
```

tambien te recomendaria generar un modelo/interfaz diferente para que regreses el token y el usuario por separado no junto ya que probablemente en el UI cuando llegue el token querras sacarlo aparte del usuario y probablemente ya no quieras recibir al usuario como respuesta, ya que llevas el id del usuario en el token y con eso lo puedes identificar para subsecuentes requests

Reply:
o sea poner la lógica de la generación del token en otro lado, no?

Reply:
si, justo eso